source/patterns/@aws-solutions-constructs/aws-fargate-secretsmanager/test/fargate-secretsmanager.test.ts

```diff
@@ -0,0 +1,639 @@
+    publicApi,
+    ecrRepositoryArn: defaults.fakeEcrRepoArn,
+    vpcProps: { cidr: '172.0.0.0/16' },
+    clusterProps: { clusterName },
+    containerDefinitionProps: { containerName },
+    fargateTaskDefinitionProps: { family: familyName },
+    fargateServiceProps: { serviceName },
+    secretProps: {
+      secretName
+    },
+  });
+
+  expect(construct.vpc !== null);
+  expect(construct.service !== null);
+  expect(construct.container !== null);
+  expect(construct.secret !== null);
+
+  expect(stack).toHaveResourceLike("AWS::ECS::Service", {
+    ServiceName: serviceName,
+    LaunchType: 'FARGATE',
+    DesiredCount: 2,
+    DeploymentConfiguration: {
+      MaximumPercent: 150,
+      MinimumHealthyPercent: 75
+    },
+    PlatformVersion: ecs.FargatePlatformVersion.LATEST,
+  });
+
+  expect(stack).toHaveResourceLike("AWS::ECS::Cluster", {
+    ClusterName: clusterName
+  });
+
+  expect(stack).toHaveResourceLike("AWS::SecretsManager::Secret", {
+    Name: secretName
+  });
+
+  expect(stack).toHaveResourceLike("AWS::IAM::Policy", {
+    PolicyDocument: {
+      Statement: [
+        {
+          Action: [
+            "secretsmanager:GetSecretValue",
+            "secretsmanager:DescribeSecret"
+          ],
+          Effect: "Allow",
+          Resource: {
+            Ref: "testconstructsecret1A43460A"
+          }
+        }
+      ]
+    }
+  });
+
+  expect(stack).toHaveResourceLike("AWS::ECS::TaskDefinition", {
+    Family: familyName,
+    ContainerDefinitions: [
+      {
+        Essential: true,
+        Image: {
+          "Fn::Join": [
+            "",
+            [
+              "123456789012.dkr.ecr.us-east-1.",
+              {
+                Ref: "AWS::URLSuffix"
+              },
+              "/fake-repo:latest"
+            ]
+          ]
+        },
+        MemoryReservation: 512,
+        Name: containerName,
+        PortMappings: [
+          {
+            ContainerPort: 8080,
+            Protocol: "tcp"
+          }
+        ]
+      }
+    ]
+  });
+
+  expect(stack).toHaveResourceLike("AWS::EC2::VPC", {
+    CidrBlock: '172.0.0.0/16'
+  });
+
+  expect(stack).toHaveResource("AWS::EC2::VPCEndpoint", {
+    ServiceName: {
+      "Fn::Join": [
+        "",
+        [
+          "com.amazonaws.",
+          {
+            Ref: "AWS::Region"
+          },
+          ".secretsmanager"
+        ]
+      ]
+    }
+  });
+
+  // Confirm we created a Public/Private VPC
+  expect(stack).toHaveResourceLike('AWS::EC2::InternetGateway', {});
+  expect(stack).toCountResources('AWS::EC2::VPC', 1);
+  expect(stack).toCountResources('AWS::SecretsManager::Secret', 1);
+  expect(stack).toCountResources('AWS::ECS::Service', 1);
+});
+
+test('New service/new secret, private API, new VPC', () => {
+  const stack = new cdk.Stack();
+  const publicApi = false;
+
+  new FargateToSecretsmanager(stack, 'test-construct', {
+    publicApi,
+    ecrRepositoryArn: defaults.fakeEcrRepoArn,
+    vpcProps: { cidr: '172.0.0.0/16' },
```

Comment:
Third time I see this string hardcoded in the code I need to jump in and say make it a constant.